blueprints/blueprint_tools.py

```diff
@@ -62,5 +92,77 @@ def upload_file():
 
     for file in files:
         filename = werkzeug.utils.secure_filename(os.path.basename(file.filename))
+        print(f"{filename=}")
         file.save(os.path.join(UPLOAD_FOLDER, filename))
     return flask.make_response({"message": "File uploaded"}, 201)
+
+
+@tools_routes.route(
+    tools_allowed_objects_json["route"],
+    methods=tools_allowed_objects_json["methods"],
+)
+def allowed_objects():
+    geode_functions.validate_request(flask.request, tools_allowed_objects_json)
+    file_extension = os.path.splitext(flask.request.json["filename"])[1][1:]
+    allowed_objects = geode_functions.list_geode_objects(
+        file_extension, flask.request.json["key"]
+    )
+
+    return flask.make_response({"allowed_objects": allowed_objects}, 200)
+
+
+@tools_routes.route(
+    tools_missing_files_json["route"],
+    methods=tools_missing_files_json["methods"],
+)
+def missing_files():
+    UPLOAD_FOLDER = flask.current_app.config["UPLOAD_FOLDER"]
+    geode_functions.validate_request(flask.request, tools_missing_files_json)
+
+    missing_files = geode_functions.missing_files(
+        flask.request.json["input_geode_object"],
+        os.path.join(UPLOAD_FOLDER, flask.request.json["filename"]),
+    )
+    has_missing_files = missing_files.has_missing_files()
+    mandatory_files = missing_files.mandatory_files
+    additional_files = missing_files.additional_files
+
+    mandatory_files_list = []
+    for mandatory_file in mandatory_files:
+        mandatory_files_list.append(os.path.basename(mandatory_file))
+
+    additional_files_list = []
+    for additional_file in additional_files:
+        additional_files_list.append(os.path.basename(additional_file))
+
+    return flask.make_response(
+        {
+            "has_missing_files": has_missing_files,
```

Comment:
Key and value are identical. Is there a way like is JS to simplify dictionary creation? Juste out of curiosity

Reply:
I looked it out, and it doesn't seem like it exists, there's discussions in the python community about implementing this, but not yet natively it seems